src/Component/Symfony/Routing/Factory/CollectionOperationRoutePathFactory.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Resource\Symfony\Routing\Factory;
+
+use Sylius\Component\Resource\Metadata\CollectionOperationInterface;
+use Sylius\Component\Resource\Metadata\HttpOperation;
+
+class CollectionOperationRoutePathFactory implements OperationRoutePathFactoryInterface
+{
+    public function __construct(private OperationRoutePathFactoryInterface $decorated)
+    {
+    }
+
+    public function createRoutePath(HttpOperation $operation, string $rootPath): string
+    {
+        $shortName = $operation->getShortName();
+
+        if ($operation instanceof CollectionOperationInterface) {
+            $path = match ($shortName) {
+                'index', 'get_collection' => '',
+                default => '/' . $shortName,
+            };
+
+            return sprintf('%s%s', $rootPath, $path);
+        }
+
+        return $this->decorated->createRoutePath($operation, $rootPath);
```

Comment:
The decoration is an interesting choice for the structure of these classes. It works, but I would argue, that perhaps some strategy-oriented approach, like OperationRoutePathFactories indexed by instance of operation would be a better choice